HiP-App/HiP-Auth-v2
# Iss hipcms 620

Test setup (this is the same as in the previous PR for 617, so @SvenEV can hopefully skip this):

* download the Identityserver4 samples from https://github.com/IdentityServer/IdentityServer4.Samples
* in the project "JsOidc" open the app.js file and set client_id: "HiP-TokenGenerator" and scope: "openid profile email HiP-CmsWebApi"
* in the `index.html` replace all "api1" strings by "HiP-CmsAngularApp"
* open the appsettings.Development.json of the CmsWebApi project and set "DOMAIN": "http://localhost:5001",
* run the JsOidc application, the Auth-v2 service and the CmsWebApi service

Testing the actual DB stuff:

* run a MSSQL database (e.g. using a docker image from here https://hub.docker.com/r/microsoft/mssql-server-linux/)
* in the `appsettings.Development.json` set the `*_DB_USERNAME` and `*_DB_PASSWORD` strings to the values that you set in the previous step
* maybe you have to run `dotnet ef database update` in the project directory now (pls report if yes)
* go to http://localhost:5001 and register as a new user by clicking the `Register` Button at the top
* go to http://localhost:7101 and request an API token (when asked for the login data use the user you just created)
* in the next window you should get a token back, copy the token value and try a authenticated request (for example http://localhost:5000/Api/Annotation/Tags) to the CmsWebApi with the Authorization Header set to Bearer [YOUR TOKEN HERE]
* you should get a status code 200 back
